### ISSUE TITLE
No Way To Pass Variables From Child To Layout

### ISSUE THREAD

**Issue by cfsilence** [ultraq/thymeleaf-layout-dialect#155]
I've written up some thoughts on this dialect - and illustrated my concerns in this blog post:

http://recursive.codes/blog/post/22

Any chance you have a plan to address this concern?

**Comment by ultraq** [comment]
Hi Todd!  First of all, nice blog - I came across it the other day when you wrote about using Thymeleaf w/ Spark 😃 

Anyway, as for passing values from child templates up to their parent layouts, it's possible using Thymeleaf's `th:with` attribute processors on any element that's involved in the layout/decoration process, which would be anywhere `layout:decorate` or `layout:fragment` will be found.  eg:

Child/content template:

```html
<html layout:decorate="your-layout.html" th:with="greeting='Hello!'">
```

Parent/layout template:

```html
<html>
  ...
  <p th:with="${greeting}"></p> <!-- You'll end up with "Hello!" in here -->
```

Now, I don't seem to have documented this anywhere, and as someone who takes some pride in writing good docs, I feel a little bad that I've missed this!  I *swear* I used to have it somewhere because others have come to me whenever this feature is broken/missing or they couldn't get it to work.
